test/overseer/store/jdbc_test.clj

```diff
@@ -0,0 +1,113 @@
+            :password ""}})]
+    (core/install store)
+    store))
+
+(deftest test-where-map->clause
+  (is (= "age = 30 AND email = 'bob@example.com'"
+         (store/where-map->clause
+           {:age 30
+            :email "bob@example.com"}))))
+
+(deftest test-job->jdbc-map
+  (let [job {:job/id "foo-id"
+             :job/type "process"
+             :job/status :unstarted
+             :job/args {:customer_id 1}}]
+    (is (= {:id "foo-id"
+            :type "process"
+            :status "unstarted"
+            :args (pr-str {:customer_id 1})}
+           (store/job->jdbc-map job)))))
+
+(deftest test-jdbc-map->job
+  (let [jdbc-map {:id "foo-id"
+                  :type "process"
+                  :status "unstarted"
+                  :args (pr-str {:customer_id 1})}]
+    (is (= {:job/id "foo-id"
+            :job/type :process
+            :job/status :unstarted
+            :job/args {:customer_id 1}}
+           (store/jdbc-map->job jdbc-map)))))
+
+(deftest test-job-jdbc-map-roundtrip
+  (let [job (test-utils/job)]
+    (is (= job (store/jdbc-map->job (store/job->jdbc-map job))))))
+
+(deftest test-update-job
+  ; Test that optimistic concurrency is implemented correctly (via lock_version)
+  ; Start two concurrent update attempts on the same row; expect only 1 to win
+  ; and subsequently update the row / increment the lock
+  (let [{:keys [db-spec] :as store} (test-store)
+        {job-id :job/id :as job} (test-utils/job)
+
+        conflicts (atom 0)
+
+        attempt-update
+        (fn []
+          (try
+            (store/update-job db-spec job-id {:status "unstarted"} {:status "started"})
+            (catch clojure.lang.ExceptionInfo ex
+              (if (= :stale-object (:type (ex-data ex)))
+                (swap! conflicts inc)
+                (throw ex)))))]
+    (core/transact-graph store (api/simple-graph job))
+    (run! deref [(future (attempt-update))
+                 (future (attempt-update))])
+    (is (= 1 @conflicts))
+    (is (= 1 (store/query-lock-version db-spec job-id)))
+    (is (= :started (:job/status (store/query-job db-spec job-id))))))
+
+(deftest test-dependents
+  (let [{:keys [db-spec] :as store} (test-store)
+
+        j0 (test-utils/job {:job/id "j0"})
+        j1 (test-utils/job {:job/id "j1"})
+        j2 (test-utils/job {:job/id "j2"})
+
+        graph
+        (loom.graph/digraph
+          {j0 []
+           j1 [j0]
+           j2 [j1]})]
+    (core/transact-graph store graph)
+    (is (= #{"j1" "j2"} (store/dependents db-spec "j0")))))
+
+(deftest test-job-dep-jdbc-maps
+  (let [j0 (test-utils/job {:job/id "j0"})
+        j1 (test-utils/job {:job/id "j1"})
+        j2 (test-utils/job {:job/id "j2"})
+
+        graph
+        (loom.graph/digraph
+          {j0 []
+           j1 [j0]
+           j2 [j1]})]
+    (is (= #{{:job_id "j1" :dep_id "j0"}
+             {:job_id "j2" :dep_id "j1"}}
+           (set (store/job-dep-jdbc-maps graph))))))
+
```

Comment:
Not a big deal, I almost kind of want a test of `(core/install)` just to make sure it guarantees that the schema got setup properly, with indexes, etc.